### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by JDTobin**
Adds permissions and permissions checking to API endpoints.

Removes knox in favor of using django sessions.

Expands password validation checks.

Adds logout endpoint to end current session.

Includes authentication options with XDS UI Configuration.

Minor update to nginx file serving.

### CODE REVIEW

File: app/core/models.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
+import re
```
Comment (rglasgow-deloitte) [inline app/core/models.py:1]: What is this import?

Comment (JDTobin) [reply app/core/models.py:1]: Regex, it is used in the password validators @KarenAJ  wrote

File: app/xds_api/views.py [hunk 2]
```diff
@@ -135,61 +142,122 @@ def get(self, request):
...
+        if username is None or password is None:
+            return JsonResponse({"info": "Username and Password is needed"},
+                                status=status.HTTP_401_UNAUTHORIZED)
+
+        # attempt login using credentials
+        user = authenticate(username=username, password=password)
+
+        # check if authentication was successful
+        if user is None:
+            return JsonResponse({"info": "User does not exist"},
```
Comment (rglasgow-deloitte) [inline app/xds_api/views.py:222]: for consistency should we be using Response over JsonResponse

File: app/core/models.py (hunk 1 above)
Comment (rglasgow-deloitte) [reply app/core/models.py:1]: Ah! looks good then.

**Review by rglasgow-deloitte — APPROVED** [review]

**Review by SHGODE — APPROVED** [review]
Approved!

**Pull request merged**